cmake/ExternalSLEPc.cmake

```diff
@@ -70,30 +72,49 @@ if(NOT "${BLAS_LAPACK_LIBRARIES}" STREQUAL "")
   )
 endif()
 
-# Configure SLEPc eigenvalue solver
-if(PALACE_WITH_SLEPC)
-  list(APPEND PETSC_OPTIONS
-    "--download-slepc"
-    "--download-slepc-commit=${EXTERN_SLEPC_GIT_TAG}"
-    "--download-slepc-configure-arguments=\"--with-feast=0\""
-    # "--download-slepc-configure-arguments=\"--with-slepc4py=1\""
-  )
-endif()
-
 string(REPLACE ";" "; " PETSC_OPTIONS_PRINT "${PETSC_OPTIONS}")
 message(STATUS "PETSC_OPTIONS: ${PETSC_OPTIONS_PRINT}")
 
 include(ExternalProject)
 ExternalProject_Add(petsc
   DEPENDS             ${PETSC_DEPENDENCIES}
-  GIT_REPOSITORY      ${CMAKE_CURRENT_SOURCE_DIR}/petsc
+  GIT_REPOSITORY      ${EXTERN_PETSC_URL}
   GIT_TAG             ${EXTERN_PETSC_GIT_TAG}
-  SOURCE_DIR          ${CMAKE_CURRENT_BINARY_DIR}/petsc
+  SOURCE_DIR          ${CMAKE_BINARY_DIR}/extern/petsc
   INSTALL_DIR         ${CMAKE_INSTALL_PREFIX}
-  PREFIX              ${CMAKE_CURRENT_BINARY_DIR}/petsc-cmake
+  PREFIX              ${CMAKE_BINARY_DIR}/extern/petsc-cmake
   BUILD_IN_SOURCE     TRUE
   UPDATE_COMMAND      ""
   CONFIGURE_COMMAND   ./configure ${PETSC_OPTIONS}
   TEST_COMMAND        ${CMAKE_MAKE_PROGRAM} check  # Use auto-detected PETSC_DIR/PETSC_ARCH
   TEST_BEFORE_INSTALL TRUE
 )
+
+
+# Configure SLEPc eigenvalue solver (most options come from PETSc)
+set(SLEPC_OPTIONS
+  "--prefix=${CMAKE_INSTALL_PREFIX}"
+  "--with-feast=0"
+  "--with-arpack=0"
+  # "--with-slepc4py=1
```

Comment:
debris?

Reply:
I'll leave this in case it becomes useful at some point.